tests/bdd/features/volume_group/create/test_create.py

```diff
@@ -0,0 +1,258 @@
+    assert "exception" in shared_context
+    assert shared_context["exception"] == "Insufficient Storage"
+
+
+@then("2 volumes should be created with 1 replica each")
+def _():
+    """2 volumes should be created with 1 replica each."""
+    filtered_volumes = get_volume_group_volumes()
+    assert len(filtered_volumes) == 2
+    for volume in filtered_volumes:
+        assert volume.spec.num_replicas == 1
+
+
+@then("3 volumes should be created with 1 replica each")
+def _():
+    """3 volumes should be created with 1 replica each."""
+    filtered_volumes = get_volume_group_volumes()
+    assert len(filtered_volumes) == 3
+    for volume in filtered_volumes:
+        assert volume.spec.num_replicas == 1
+
+
+@then("3 volumes should be created with 3 replica each")
+def _():
+    """3 volumes should be created with 3 replica each."""
+    filtered_volumes = get_volume_group_volumes()
+    assert len(filtered_volumes) == 3
+    for volume in filtered_volumes:
+        assert volume.spec.num_replicas == 3
+
+
+@then("each replica of individual volume should be on a different node")
+def _():
+    """each replica of individual volume should be on a different node."""
+    filtered_volumes = get_volume_group_volumes()
+    for volume in filtered_volumes:
+        replica_topology = volume["state"]["replica_topology"]
+        node_locations = [replica_topology[uuid]["node"] for uuid in replica_topology]
+        assert len(set(node_locations)) == len(replica_topology)
+
+
+@then("each replica should reside on a different node")
+def _():
+    """each replica should reside on a different node."""
+    filtered_volumes = get_volume_group_volumes()
+    already_encountered_nodes = set()
+    for volume in filtered_volumes:
+        replica_topology = volume["state"]["replica_topology"]
+        for uuid in replica_topology:
+            node = replica_topology[uuid]["node"]
+            if node in already_encountered_nodes:
+                assert False
+            already_encountered_nodes.add(node)
+
+
+@then("each replica should reside on a different pool")
+def _():
+    """each replica should reside on a different pool."""
+    filtered_volumes = get_volume_group_volumes()
+    already_encountered_pools = set()
+    for volume in filtered_volumes:
+        replica_topology = volume["state"]["replica_topology"]
+        for uuid in replica_topology:
+            pool = replica_topology[uuid]["pool"]
+            if pool in already_encountered_pools:
+                assert False
+            already_encountered_pools.add(pool)
+
+
+@then("pools can have more than one replicas all belonging to different volumes")
+def _():
+    """pools can have more than one replicas all belonging to different volumes."""
+    filtered_volumes = get_volume_group_volumes()
+    pool_replicas = {}
+    for volume in filtered_volumes:
+        replica_topology = volume["state"]["replica_topology"]
```

Comment:
can we not use volume.state.replica_topology ?

Reply:
Interesting, this did not show up in ide, thats why missed it. Anyway fixed it now.